internal/pkg/cli/cli.go

```diff
@@ -12,62 +12,23 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/copilot-cli/internal/pkg/term/color"
-	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
 	"github.com/aws/copilot-cli/internal/pkg/workspace"
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
 )
 
-// GlobalOpts holds fields that are used across multiple commands.
-type GlobalOpts struct {
-	appName string
-	prompt  prompter
-}
```

Comment:
Curious if it's possible to keep GlobalOpts and therefore maintain DRYness across all the command Opts but just get rid of the `bindAppName` logic?

Reply:
`GlobalOpts` itself is not great because it resulted in commands getting a `prompt` whether they needed it or not, and we had a bunch of commands that never used it :(

An alternative that doesn't force the re-declaration of the `--app` flag in these sub-commands that I can think of is this: https://gist.github.com/efekarakus/37144a1a4b8792508c39e9927a020243 

The benefit is that we don't have to remember to add the flag for all these sub-commands.
But, the downside is that I find this harder to understand as a reader than:
```
cmd.Flags().StringVarP(&vars.appName, appFlag, appFlagShort, tryReadingAppName(), appFlagDescription)
```
This makes it obvious if the `--app` flag is part of the command or not.